tests/fsharp/nunitConf.fs

```diff
@@ -0,0 +1,379 @@
+                | None -> genericError "environment variable 'FSCBinPath' is required to be a valid directory, is not set"
+                | Some dir -> genericError (sprintf "environment variable 'FSCBinPath' is required to be a valid directory, but is '%s'" dir)
+            }
+
+        let smokeTest () = processor {
+            let tempFile ext = 
+                let p = Path.ChangeExtension( Path.GetTempFileName(), ext)
+                File.AppendAllText (p, """printfn "ciao"; exit 0""")
+                p
+
+            let tempDir = Commands.createTempDir ()
+            let exec exe args =
+                log "%s %s" exe args
+                use toLog = redirectToLog ()
+                Process.exec { RedirectError = Some toLog.Post; RedirectOutput = Some toLog.Post; RedirectInput = None } tempDir cfg.EnvironmentVariables exe args
+
+            do! Commands.fsc exec cfg.FSC "" [ tempFile ".fs" ] |> checkResult
+
+            do! Commands.fsi exec cfg.FSI "" [ tempFile ".fsx" ] |> checkResult
+        
+            }
+    
+        do! checkfscBinPath ()
+
+        do! smokeTest ()
+
+        return cfg
+    } 
+
+
+let suiteHelpers = lazy (
+    initializeSuite ()
+    |> Attempt.Run 
+    |> function Success x -> x | Failure err -> failwith (sprintf "Error %A" err)
+)
+
+[<AttributeUsage(AttributeTargets.Assembly)>]
+type public InitializeSuiteAttribute () =
+    inherit TestActionAttribute()
+
+    override x.BeforeTest details =
+        if details.IsSuite 
+        then suiteHelpers.Force() |> ignore
+
+    override x.AfterTest details =
+        ()
+
+    override x.Targets with get() = ActionTargets.Test ||| ActionTargets.Suite
+
+
+[<assembly:InitializeSuite()>]
+()
+
+module FSharpTestSuite =
+
+    let getTagsOfFile path =
+        match File.ReadLines(path) |> Seq.take 5 |> Seq.tryFind (fun s -> s.StartsWith("// #")) with
+        | None -> []
+        | Some line -> 
+            line.TrimStart('/').Split([| '#' |], StringSplitOptions.RemoveEmptyEntries)
+            |> Seq.map (fun s -> s.Trim())
+            |> Seq.filter (fun s -> s.Length > 0)
+            |> Seq.distinct
+            |> Seq.toList
+
+    let getTestFileMetadata dir =
+        Directory.EnumerateFiles(dir, "*.fs*")
+        |> Seq.toList
+        |> List.collect getTagsOfFile
+
+    let parseTestLst path =
+        let dir = Path.GetDirectoryName(path)
+        let commentLine (t: string) = t.StartsWith("#")
+        let lines =
+            File.ReadAllLines(path)
+            |> Array.filter (not << commentLine)
+            |> Array.filter (not << String.IsNullOrWhiteSpace)
+        let parse (t: string) =
+            let a = t.Split([| '\t'; '\t' |], StringSplitOptions.RemoveEmptyEntries)
+            let testDir = Commands.getfullpath dir a.[1]
+            [| for x in a.[0].Split(',') do yield (x, testDir) |]
+
+        lines |> Array.collect parse |> List.ofArray
+
+    let ``test.lst`` = lazy ( 
```

Comment:
that's done for every test ( i need to read categories ) so i need a cache, lazy it's easy.